Documents/Command/AWS-RunAnsible/AWS-RunAnsible.yaml

```diff
@@ -0,0 +1,189 @@
+  (including any roles and custom modules). Use PlaybookFile for the playbook to run
+  from that URL. Use the extravar parameter to send runtime variables to the Ansible
+  execution. Use the check parameter to perform a dry run of the Ansible execution.
+  The output of the dry run shows the changes that will be made when the playbook
+  is executed. This version also supports zip files via http or s3 URLs. It also supports
+  providing an s3 URL for a directory containing the playbook files. '
+parameters:
+  playbook:
+    type: String
+    description: (Optional) If you don't specify a URL, then you must specify playbook
+      YAML in this field.
+    default: ''
+    displayType: textarea
+  PlaybookURL:
+    type: String
+    description: '(Optional) If you don''t specify playbook YAML, then you must specify
+      a URL where the playbook is stored. This url can refer to a playbook yaml file,
+      an s3 prefix, or an archive (TAR GZIP/BZIP2, Zip) containing all your playbook
+      artifacts. You can specify the URL in the following formats: http://example.com/playbook.yml
+      or s3://examplebucket/playbook.url. You can also specify an s3 URL for a directory
+      containing the playbooks like s3://mybucket/myplabooks/.  For security reasons,
+      you can''t specify a URL with quotes.'
+    default: ''
+    allowedPattern: ^\s*$|^(http|https|s3)://[^']*$
+  PlaybookFile:
+    type: String
+    description: (Optional) The playbook file to run (including relative path), when
+      specifying a playbook bundle like a zip file or an s3 path to a folder in a
+      bucket
+    default: ssm-playbook.yml
+  extravars:
+    type: String
+    description: '(Optional) Additional variables to pass to Ansible at runtime. Enter
+      a space separated list of key/value pairs. For example: color=red flavor=lime'
+    default: SSM=True
+    displayType: textarea
+    allowedPattern: ^$|^\w+\=[^\s|:();&]+(\s\w+\=[^\s|:();&]+)*$
+  WorkingDirectory:
+    type: String
+    description: ' (Optional) Current working directory to use for execution; if not
+      specified, use random temp directory.'
+    allowedPattern: ^\s*$|^[^']+$
+    default: ''
+  check:
+    type: String
+    description: ' (Optional) Use the check parameter to perform a dry run of the
+      Ansible execution.'
+    allowedValues:
+    - 'True'
+    - 'False'
+    default: 'False'
+  verbose:
+    type: String
+    description: ' (Optional) Set the level of verbosity for logging the execution
+      of the playbook. -v for low verbosity, -vvvv for debug level.'
+    allowedValues:
+    - -v
+    - -vv
+    - -vvv
+    - -vvvv
+    default: -v
+mainSteps:
+- action: aws:runShellScript
+  name: runShellScript
+  inputs:
+    runCommand:
+    - '#!/bin/bash'
+    - 'echo "Installing and or updating required tools: Ansible, wget unzip ...."
+      >&2'
+    - 'if cat /etc/issue|grep -i Amazon ; then '
+    - '   sudo pip install ansible --upgrade'
```

Comment:
The installed version of pip and any other executable whose name changes with different versions should be detected. For example, Ubuntu 16.04 comes with (only) Python 3.5 by default and the pip executable is "pip3" not "pip". Here's an example you can (test and) use:

```bash
#!/bin/bash
if command -v pip &> /dev/null; then
    PIP=pip
elif command -v pip2 &> /dev/null; then
    PIP=pip2
elif command -v pip3 &> /dev/null; then
    PIP=pip3
else
    echo "Failed to find Python package manager (pip)!"
    exit -1
fi
echo "pip = '$PIP'"
```

Related to [issue 26](https://github.com/awslabs/aws-systems-manager/issues/26) regarding unsafe Python version assumptions.